src/Simplify_Max.cpp

```diff
@@ -173,6 +173,22 @@ Expr Simplify::visit(const Max *op, ExprInfo *bounds) {
              rewrite(max(x, max(min(x, y), z)), max(x, z)) ||
              rewrite(max(x, max(min(y, x), z)), max(x, z)) ||
 
+             rewrite(max(select(x, y, z), z), select(x, max(y, z), z)) ||
```

Comment:
@jn80842 I believe this rule is OK because we prefer stronger IR nodes at the root of the true, all else equal. Is that right?

Reply:
Ok, according to Julie this rule is actually not OK because we prefer weaker nodes at the root, so this may fight with another rule that moves a max inside a select for some reason.

Reply:
Hmm, I'm going to have to look at this, because I did check that these rules are needed, and they really are. I guess I'll have to try to find a more specific rule.